google/cloud/firestore_v1/aggregation.py

```diff
@@ -70,20 +75,37 @@ def get(
                 should be retried.  Defaults to a system-specified policy.
             timeout (float): The timeout for this request.  Defaults to a
                 system-specified value.
+            explain_options
+                (Optional[:class:`~google.cloud.firestore_v1.query_profile.ExplainOptions`]):
+                Options to enable query profiling for this query. When set,
+                explain_metrics will be available on the returned generator.
 
         Returns:
             list: The aggregation query results
 
         """
-        result = self.stream(transaction=transaction, retry=retry, timeout=timeout)
-        return list(result)  # type: ignore
+        result = self.stream(
+            transaction=transaction,
+            retry=retry,
+            timeout=timeout,
+            explain_options=explain_options,
+        )
+        result_list = list(result)
 
-    def _get_stream_iterator(self, transaction, retry, timeout):
+        if explain_options is None:
+            explain_metrics = None
+        else:
+            explain_metrics = result.explain_metrics
+
```

Comment:
Indeed this same if/else code appears repeatedly in all get() methods. But considering we want to expand DocumentSnapshotList into a more general list class that may encompass types other than DocumentSnapshot, I wonder if the same logic checking explain_options will apply to all future classes that form the list? Also if we put this logic inside DocumentSnapshotList, we would need to pass the original `result` (StreamGenerator) into the initialization. I personally feel like it's cleaner to initialize with the explain_metrics, but I also want to hear your opinion on this.

Reply:
On second thought, I think you're right, it's probably better to leave most of this stuff outside the __init__

I was thinking it could be cleaner to just pass in a `StreamGenerator` on __init__, since it should contain all the information needed by `DocumentSnapshotList`. But that gets complicated on the async side, because __init__ isn't an async function. So we'd have trouble building the object there

Still though, one thing to consider: if we pass in the `StreamGenerator` and hold a reference to it in `DocumentSnapshotList`, you can delegate to that instance when calling `explain_metrics()`, which could help reduce some duplication

Reply:
When I'm trying to implement this, I still feel a bit unsure. Passing `StreamGenerator` feels like we are coupling it with `QueryResultsList` really deeply and may make these two classes less versatile in the future.

Reply:
If it seems messy, don't worry about it, I was just throwing out ideas. We can just re-visit this if the duplicated code expands in the future

IMO I think it makes sense for `StreamGenerator` and `QueryResultsList` to be coupled in some way though, either through inheritance or composition. They have essentially the same use-case, QueryResultsList just has the stream fully loaded into memory